hbase-client/src/main/java/org/apache/hadoop/hbase/exceptions/ClientExceptionsUtil.java

```diff
@@ -59,18 +58,22 @@ public static boolean isMetaClearingException(Throwable cur) {
     if (cur == null) {
       return true;
     }
-    return !isSpecialException(cur) || (cur instanceof RegionMovedException)
-        || cur instanceof NotServingRegionException;
+    return !regionDefinitelyOnTheRegionServerException(cur);
   }
 
-  public static boolean isSpecialException(Throwable cur) {
-    return (cur instanceof RegionMovedException || cur instanceof RegionOpeningException
-        || cur instanceof RegionTooBusyException || cur instanceof RpcThrottlingException
-        || cur instanceof MultiActionResultTooLarge || cur instanceof RetryImmediatelyException
-        || cur instanceof CallQueueTooBigException || cur instanceof CallDroppedException
-        || cur instanceof NotServingRegionException || cur instanceof RequestTooBigException);
+  private static boolean regionDefinitelyOnTheRegionServerException(Throwable t) {
+    return (t instanceof RegionTooBusyException || t instanceof RpcThrottlingException
+        || t instanceof RetryImmediatelyException || t instanceof CallQueueTooBigException
```

Comment:
And for CallQueueTooBigException, CallDroppedException, RequestTooBigExcetion, it does not mean that the region is on the region server? We just do not want to push too much pressure on meta so we do not want to retry on these exceptions everytime, but the method name here is a bit confusing...